4.8.7 add build qt script

fbudin69500 opened the pull request:
Tests in progress on CircleCI.

fbudin69500 committed:
ENH: Addition of a script to compile Qt on Linux & MacOS

Instructions to build Qt were available in Build-qt.txt but not
as a simple script to run. A new bash script to created to
compile and install Qt as well as zlib and OpenSSL which are
required to build Qt correctly for 3D Slicer.

fbudin69500 committed:
ENH: Addition of tests

Testing compilation script for linux using a docker
image and CircleCI.

jcfr commented:
Outstanding :+1: Thankss

jcfr commented:
I will enable circle ci with this repo

jcfr commented:
For reference: https://circleci.com/gh/fbudin69500/qt-easy-build

jcfr commented:
It is great, we will then edit https://gist.github.com/jcfr/9513568 to reference your new script :)

jcfr commented on Build-qt.sh:
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
```
As suggested [here](https://github.com/KitwareMedical/TubeTK-DashboardScripts/pull/88#discussion_r71071092), what do you think if the script is named `linux_build_qt.sh` ?

jcfr replied:
Never-mind, since the script handle both MacOSx and Linux, you can ignore.

Sorry for the noise

jcfr commented on Build-qt.sh:
```diff
@@ -0,0 +1,180 @@
+then
+  # MacOS
+  deployment_target=10.6
+  sysroot=/Developer/SDKs/MacOSX10.6.sdk
```
The deployment_target and sysroot should be configurable

jcfr replied:
Logic available here could be re-used/adapted .. https://github.com/KitwareMedical/TubeTK/blob/master/CMake/TubeTKBlockSetCMakeOSXVariables.cmake

fbudin69500 replied:
deployment_target, sysroot, and architecture are now options.  It would be indeed great to adapt the CMake code you are mentioning above, but I don't have a Mac to test the script...

jcfr replied:
Thanks.

If you would like, from the internal network, you could connect to `10.171.2.166`, this is the MacOSX `factory-south`

fbudin69500 commented:
@jcfr : thanks for the review

jcfr commented:
Nice.

To test on MacOSx, `.travis.yml` could be used.

Here is an example showing how to enable osx: https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/blob/c25eb7c4bc4e18050e91129f81693b7d5c32f503/.travis.yml#L3

fbudin69500 committed:
ENH: Addition of tests on TravisCI

TravisCI allows to run tests on MacOS. Tests are run directly on
the computer, not in a docker container.

jcfr commented:
Thanks again for the improvements :+1:
